Expensify/App
# Fix particpant related and searching behavior for default rooms.

@HorusGoul, please review when you get the chance
CC: @yuwenmemon 

### Details
Stop showing participants of a defaultRoom when trying to create a new Chat, Group, Payment Request, or Bill Split.
Also make it so that when we are in Search, using the email/name of a user doesn't bring up all the defaultRooms that user is a participant in.

### Fixed Issues

$ https://github.com/Expensify/Expensify/issues/169139

### Tests
Tested the QA Steps locally

### QA Steps
Since this is defaultRooms related, this needs to be QA'ed internally (I think?). I'll test this when this hits staging. Just ping @TomatoToaster, if this hasn't been checked off in QA checklist.
1. Click + and create a new chat, a new group, a new money request, and a new split bill and verify when the particpant selecting screen shows up, you do not see any defaultRooms in there.
2. Open the search and type in policy names, domain names, and room names. Verify that they show the default rooms and that those options always come first in the search *once you type something in*. It should still show the recents until you type something.
3. Verify that searching for a participant's name doesn't bring up the default rooms they are in.


### Tested On


### Screenshots


#### Web


#### Mobile Web


#### Desktop


#### iOS


#### Android
